Commands/Search.cs

```diff
@@ -0,0 +1,111 @@
+﻿
+//System
+using System;
+using System.Threading.Tasks;
+
+//Discord 
+using Discord.Commands;
+
+//Google 
+using static Google.Apis.Customsearch.v1.CseResource;
+using static Google.Apis.Customsearch.v1.CseResource.ListRequest;
+using Google.Apis.Services;
+using Google.Apis.Customsearch.v1;
+using Discord;
+
+namespace SteamDiscordBot.Commands
+{
+    public class Search : ModuleBase
+    {
+        static SafeEnum SafeSearch;
+        static int Search_Number;
+        static readonly string[] FLAGS = new string[] { "-safesearch_active", "-safesearch_off", "-num"};
+        
+        [Command("search"), Summary("Peforms google search and returns result(s)")]
+        public async Task Say(params string[] args)
+        {
+            try
+            {
+                CustomsearchService service = new CustomsearchService(new BaseClientService.Initializer { ApiKey = Program.config.GoogleApiKey });
```

Comment:
We assume that `Program.config.GoogleApiKey` exists, but older configurations won't have this. If we kept this as-is, older settings.json would fail when this command is executed. Can we check if this property exists first, and if it doesn't display a warning out and return? Look around the codebase for other bcompat checks, there's already a method implemented to determine if GoogleApiKey is a valid member of Program.config

Reply:
I've given this more thought, we can `Log` an output to the console, and then also message in response and say something like `This command must have a valid API Key supplied! Contact the Owner`

Hope that makes this more clear